src/main/java/org/mule/extension/file/internal/DirectoryListener.java

```diff
@@ -277,7 +277,10 @@ private void listen(SourceCallback<InputStream, ListenerFileAttributes> sourceCa
         }
       }
     } catch (Exception e) {
-      sourceCallback.onSourceException(e);
+      LOGGER.error(format("Directory listener at flow '%s' found exception listening for File System changes. "
```

Comment:
if that happens you'll get it as a notification. Anyway, remember this class is not productive. We're just keeping it for future evaluation